src/probnum/linops/_linear_operator.py

```diff
@@ -416,40 +390,53 @@ class ScalarMult(LinearOperator):
         Scalar to multiply by.
     """
 
-    def __init__(self, shape, scalar):
-        self.scalar = scalar
-        super().__init__(shape=shape, dtype=float)
+    def __init__(
+        self,
+        shape: ShapeArgType,
+        scalar: ScalarArgType,
+        dtype: Optional[DTypeArgType] = None,
+    ):
+        self._scalar = probnum.utils.as_numpy_scalar(scalar, dtype=dtype)
 
-    def _matvec(self, x):
-        return self.scalar * x
+        super().__init__(shape=probnum.utils.as_shape(shape), dtype=self._scalar.dtype)
 
-    def _matmat(self, X):
-        return self.scalar * X
+    @property
+    def scalar(self):
```

Comment:
Not crucial in this PR, but some type hints for these methods are missing. Feel free to ignore for now.

Reply:
Yeah, I have that in my TODO and will follow up on type hints (and verify them by tests) in one of the next PRs